beetle.gemspec

```diff
@@ -30,7 +30,6 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency "amqp",                    "= 1.8.0"
   s.add_runtime_dependency "activesupport",           ">= 2.3.4"
 
-  s.add_development_dependency "uuid4r",              ">= 0.1.2"
```

Comment:
Without this removal the other changes could be considered non-breaking, couldn't they?

Reply:
well, adding a new callback seems an API change to me

Reply:
And we consider it breaking because somebody could have already implemented a `post_process` or `pre_process` method for another purpose?

Reply:
That's one breakage. The other is: I need to adapt xing-amqp to make use of the new functionality. And then it relies on it.